native/templates/ios/AppDelegate.mm

```diff
@@ -55,11 +56,8 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     [self.window setRootViewController:_viewController];
 
     [self.window makeKeyAndVisible];
-
-    _platform = dynamic_cast<cc::IOSPlatform *>(cc::BasePlatform::getPlatform());
-    CCASSERT(_platform != nullptr, "Platform pointer can't be null");
-    _platform->loop();
-
+    
+    [[SDKWrapper shared] application:application didFinishLaunchingWithOptions:launchOptions];
```

Comment:
Do we have to re-use the logic of SDKWrapper? It seems that SDKWrapper and SDKDelegate should only take care of SDK things, however, now we use it for notifying app lifecycle event to engine too.

Reply:
Yes, but think in another way, cocos is also an sdk for users. If the mechanism of SDKWrapper is not used, the notification event will be sent twice

Reply:
I think we can keep the same logic as before before we have a good design.

Reply:
OK

Reply:
Yes, I agree with what @minggo said.
We need to treat SDKWrapper as a standalone 3rd SDK.

https://github.com/cocos/cocos-engine/blob/v3.5.0/native/cocos/platform/ios/AppDelegate.mm#L72

I think it's ok to keep the old logic , just move AppDelegate.mm out of engine.

Reply:
Or
Is it better to add an `objective-c` adaptor for synchronize system event to engine rather than use `_platform` C++ object directly?

Using `objective-c` code will be more easily to transfer event to engine. It's will be similar with what SDKWrapper does.

AppDelegate.mm
```objc
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
    [[SDKWrapper shared] application:application didFinishLaunchingWithOptions:launchOptions];
    // Add the view controller's view to the window and display.
    CGRect bounds = [[UIScreen mainScreen] bounds];
    self.window   = [[UIWindow alloc] initWithFrame:bounds];

    // Should create view controller first, cc::Application will use it.
    _viewController                           = [[ViewController alloc] init];
    _viewController.view                      = [[View alloc] initWithFrame:bounds];
    _viewController.view.contentScaleFactor   = UIScreen.mainScreen.scale;
    _viewController.view.multipleTouchEnabled = true;
    [self.window setRootViewController:_viewController];

    [self.window makeKeyAndVisible];

    [[PlatformEventAdapter sharedAdapter] application:application didFinishLaunchingWithOptions:launchOptions];
    return YES;
}

- (void)applicationWillResignActive:(UIApplication *)application {
    [[SDKWrapper shared] applicationWillResignActive:application];
    [[PlatformEventAdapter sharedAdapter] applicationWillResignActive:application];
}

- (void)applicationDidBecomeActive:(UIApplication *)application {
    [[SDKWrapper shared] applicationDidBecomeActive:application];
    [[PlatformEventAdapter sharedAdapter] applicationDidBecomeActive:application];
}
```

// ViewController.mm
```objc
- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
    [[PlatformEventAdapter sharedAdapter] viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
}
```

// PlatformEventAdapter.mm
```objc
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {

    _platform = dynamic_cast<cc::IOSPlatform *>(cc::BasePlatform::getPlatform());
    CCASSERT(_platform != nullptr, "Platform pointer can't be null");
    _platform->loop();

    return YES;
}

- (void)applicationWillResignActive:(UIApplication *)application {
    _platform->onPause();
}

- (void)applicationDidBecomeActive:(UIApplication *)application {
    _platform->onResume();
}

- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
    cc::IScreen::Orientation orientation;
    // reference: https://developer.apple.com/documentation/uikit/uiinterfaceorientation?language=objc
    // UIInterfaceOrientationLandscapeRight = UIDeviceOrientationLandscapeLeft
    // UIInterfaceOrientationLandscapeLeft = UIDeviceOrientationLandscapeRight
    switch ([UIDevice currentDevice].orientation) {
        case UIDeviceOrientationPortrait:
            orientation = cc::IScreen::Orientation::PORTRAIT;
            break;
        case UIDeviceOrientationLandscapeRight:
            orientation = cc::IScreen::Orientation::LANDSCAPE_LEFT;
            break;
        case UIDeviceOrientationPortraitUpsideDown:
            orientation = cc::IScreen::Orientation::PORTRAIT_UPSIDE_DOWN;
            break;
        case UIDeviceOrientationLandscapeLeft:
            orientation = cc::IScreen::Orientation::LANDSCAPE_RIGHT;
            break;
        default:
            break;
    }
    cc::DeviceEvent ev;
    cc::BasePlatform* platform = cc::BasePlatform::getPlatform();
    cc::IScreen* screenIntf = platform->getInterface<cc::IScreen>();
    ev.type           = cc::DeviceEvent::Type::DEVICE_ORIENTATION;
    ev.args[0].intVal = static_cast<int>(screenIntf->getDeviceOrientation());
    AppDelegate* delegate = [[UIApplication sharedApplication] delegate];
    [delegate dispatchEvent:ev];

    float pixelRatio = screenIntf->getDevicePixelRatio();
    cc::WindowEvent resizeEv;
    resizeEv.type = cc::WindowEvent::Type::RESIZED;
    resizeEv.width = size.width * pixelRatio;
    resizeEv.height = size.height * pixelRatio;
    [delegate dispatchEvent:resizeEv];
    //CAMetalLayer is available on ios8.0, ios-simulator13.0.
    CAMetalLayer *layer = (CAMetalLayer *)self.view.layer;
    CGSize tsize             = CGSizeMake(static_cast<int>(size.width * pixelRatio),
                                         static_cast<int>(size.height * pixelRatio));
    layer.drawableSize = tsize;
}
```

Reply:
If developers integrate cocos into their own apps as a so, isn't it like an SDK?
I will restore the old one first, but reserve my opinion

Reply:
-  iOS does not support .so. 
- SDKWrapper is only served as builtin services, it is not a universal sdk wrapper. Developers will not use it to integrate other sdk.